include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
                 remainder = exact_remainder.digits;
             }
 
-            /// calculates *this / divisor 
-            /// 
-            ///  @brief a binary-search type method for dividing exact_numbers.
-            ///  @param is_upper true: returns result with an error of +epsilon, while
-            ///                  false: returns result with an error of -epsilon
-            void divide_vector(exact_number<T> divisor, unsigned int maximum_precision) {
-                /// @TODO: replace this with something more efficient, like newton-raphson method
-                // it also completely recalculates on each precision increase
-                // instead, could use previous information to make better "guesses"
-                // for our iteration scheme.
-
-                boost::real::exact_number<T> numerator;
-                boost::real::exact_number<T> left;
-                boost::real::exact_number<T> right;
-                boost::real::exact_number<T> residual;
-                boost::real::exact_number<T> tmp;
-                boost::real::exact_number<T> half;
-                boost::real::exact_number<T> distance;
-                boost::real::exact_number<T> min_boundary_n;
-                boost::real::exact_number<T> min_boundary_p;
-
-                bool positive = ((*this).positive == divisor.positive);
-                numerator = (*this).abs();
-                divisor = divisor.abs();
-
-                // ensuring that assignment from -1 * (maximum_precision) to exponent will not
-                // overflow
+            /*              DIVISION
+             *  @brief:  divides (*this) by divisor
+             *  @param:  divisor: number which divides (*this)
+             *  @param:  maximum_precision: Absolute Error in the result should be < 1*base^(-maximum_precision)
```

Comment:
This sound incorrect, the maximum precision should be adjusted by  how many digits are before the "point", otherwise you cannot have contorl over arbitrarily large numbers.
The maximum precision goal is to have a way to constraint how much computation power can be invested in a calculation. We discussed in the past that maximum_precision may not be a the happiest of the names to express that.

Reply:
Yes, I should change the name as it is a bit confusing. It's not the maximum_precision of the const_precision_iterator. It's the no of iterations the iterator has made till now, therefore at each iteration, the maximum allowed error becomes smaller and smaller.